projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,261 @@
+				'scope'        => array(),
+				'tax_input'    => array(
+					'wp_theme' => get_option( 'stylesheet' ),
+				),
+			);
+			wp_insert_post( $template );
+		}
+	}
+
+	/**
+	 * Adds modal with Subscribe Modal content.
+	 *
+	 * @return void
+	 */
+	public function wpcom_add_subscribe_modal_to_frontend() {
+		if ( $this->is_front_end_single_post() ) {
+			$posts = get_posts(
+				array(
+					'post_type'   => 'wp_template_part',
+					'post_status' => 'publish',
+					'numberposts' => -1,
+					'post_name'   => $this->get_subscribe_template_slug(),
+				)
+			);
+
+			if ( ! $posts ) {
+				return;
+			}
+
+			$subscribe_template = end( $posts );
+
+			$blocks = parse_blocks( $subscribe_template->post_content );
+
+			/*
+			 * Jetpack requires escaping html for the render_block() call below.
+			 * I'm using wp_kses() and specifying the allowed tags.
+			 * But this is fragile: users can edit the Subscribe Modal
+			 * template and add tags that are not allowed here.
+			 */
+			$allowed_html = array(
+				'div'    => array(
+					'class' => array(),
+					'style' => array(),
+				),
+				'h2'     => array(
+					'class' => array(),
+					'style' => array(),
+				),
+				'p'      => array(
+					'type'  => array(),
+					'class' => array(),
+					'style' => array(),
+					'id'    => array(),
+				),
+				'form'   => array(
+					'action'                 => array(),
+					'method'                 => array(),
+					'accept-charset'         => array(),
+					'data-blog'              => array(),
+					'data-post_access_level' => array(),
+					'id'                     => array(),
+				),
+				'label'  => array(
+					'id'    => array(),
+					'for'   => array(),
+					'class' => array(),
+				),
+				'input'  => array(
+					'id'          => array(),
+					'type'        => array(),
+					'name'        => array(),
+					'value'       => array(),
+					'class'       => array(),
+					'style'       => array(),
+					'placeholder' => array(),
+					'required'    => array(),
+				),
+				'button' => array(
+					'type'  => array(),
+					'class' => array(),
+					'style' => array(),
+					'name'  => array(),
+				),
+			);
+
+			?>
+				<div class="jetpack-subscribe-modal">
+					<div class="jetpack-subscribe-modal__modal-content">
+						<?php
+						foreach ( $blocks as $block ) {
+							echo wp_kses( render_block( $block ), $allowed_html );
+						}
+						?>
+						<Dashicon icon="minus" className="jp-idc__idc-screen__card-action-separator" />
+					</div>
+				</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Enqueues JS to load modal.
+	 *
+	 * @return void
+	 */
+	public function wpcom_enqueue_subscribe_modal_assets() {
+		if ( $this->is_front_end_single_post() ) {
+			wp_enqueue_style( 'subscribe-modal-css', plugins_url( 'subscribe-modal.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-modal-js', plugins_url( 'subscribe-modal.js', __FILE__ ), array( 'wp-components' ), JETPACK__VERSION, true );
```

Comment:
Do we need `wp-components` to be loaded here? Could we do without it, to avoid loading too many additional things on the frontend?

Reply:
This would be important to solve — can be a follow-up PR.

Reply:
Actually we shouldn't need it, looking at `subscribe-modal.js`, but we do need `domReady` which is a tiny script.